products/endpoints/backend/api.py

```diff
@@ -128,14 +143,21 @@ def create(self, request: Request, *args, **kwargs) -> Response:
         self.validate_request(data, strict=True)
 
         try:
-            endpoint = Endpoint.objects.create(
-                team=self.team,
-                created_by=cast(User, request.user),
-                name=cast(str, data.name),  # verified in validate_request
-                query=cast(Union[HogQLQuery, InsightQueryNode], data.query).model_dump(),
-                description=data.description or "",
-                is_active=data.is_active if data.is_active is not None else True,
-            )
+            # Handle both User and ProjectSecretAPIKeyUser creators
+            create_kwargs = {
+                "team": self.team,
+                "name": cast(str, data.name),  # verified in validate_request
+                "query": cast(Union[HogQLQuery, InsightQueryNode], data.query).model_dump(),
+                "description": data.description or "",
+                "is_active": data.is_active if data.is_active is not None else True,
+            }
+
+            if isinstance(request.user, ProjectSecretAPIKeyUser):
+                create_kwargs["created_by_api_key"] = request.user.project_secret_api_key
```

Comment:
This is fundamentally my issue with this feature overall. We're going to end up with this kind of split everywhere in the codebase... Like if we have this here we will end up needing it everywhere else too. 

I think we need to decide either to _not_ have any user concept and as such have dedicated endpoints that work with projectsecret keys (like the local eval endpoint, this one maybe) _or_ we need to have each team have a special real User model that gets created associated with it for this kind of stuff (which feels much harder)

Need to think a little deeper on this...